doc/build/snippets/writing.rst

```diff
@@ -156,6 +156,25 @@ This :file:`snippet.yml` adds :file:`foo.conf` to the build:
 The path to :file:`foo.conf` is relative to the directory containing
 :file:`snippet.yml`.
 
+``.cmake`` files
+****************
+
+This :file:`snippet.yml` adds :file:`foo.cmake` to the build:
+
+.. code-block:: yaml
+
+   name: foo
+   cmake:
+     EXTRA_CMAKE_FILE: foo.cmake
+
+The path to :file:`foo.cmake` is relative to the directory containing
+:file:`snippet.yml`. Adding cmake files like this can be useful for
+extending the configuration options available in devicetree. e.g.
+
+.. code-block:: cmake
+
+   set(DTS_EXTRA_CPPFLAGS "-DMY_DTS_CONFIGURE ${DTS_EXTRA_CPPFLAGS}" PARENT_SCOPE)
```

Comment:
I see the usefulness but I feel the solution proposed is too open ended (users can do anything in such cmake file) compared to the requested functionality of being able to extend `DTS_EXTRA_CPPFLAGS`.

The fact that the solution relies on `PARENT_SCOPE` further means that the solution may put restrictions on our ability to refactor / improve our current Zephyr CMake module infrastructure in future.

Because we already have a dedicated `DTS_EXTRA_CPPFLAGS` setting in the build system, then I would much prefer that we simply extend the `snippet.yml` to also allow that setting:
```
append:
  EXTRA_CONF_FILE: foo.conf
  EXTRA_DTC_OVERLAY_FILE: foo.overlay
  DTS_EXTRA_CPPFLAGS: -DFOO
```

That also follows the principle that anything you can do with a snippet you can also do directly by hand.
(The CMake file can in principle do so much more that a single `set()`.)

Reply:
That makes sense, I've only ever used this for `DTS_EXTRA_CPPFLAGS `